lib/livebook/runtime/k8s.ex

```diff
@@ -24,18 +23,20 @@ defmodule Livebook.Runtime.K8s do
           pod_name: String.t() | nil
         }
 
-  use GenServer, restart: :temporary
-
-  require Logger
-
-  alias Livebook.K8s.Pod
+  @type config :: %{
+          context: String.t(),
+          namespace: String.t(),
+          docker_tag: String.t(),
+          pod_template: String.t(),
+          pvc_name: String.t() | nil
+        }
 
   @doc """
   Returns a new runtime instance.
   """
-  @spec new(config :: map(), req :: Req.Request.t()) :: t()
-  def new(config, req) do
```

Comment:
@mruoss I missed that before, I think preferably we don't want to pass the whole req struct when building the runtime, and rather encapsulate the necessary logic within the runtime itself (even if we do the same elsewhere) :)

Reply:
I see. Makes sense.